GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Submodules.cs

```diff
@@ -193,6 +176,29 @@ string GetSubmoduleItemImage(DetailedSubmoduleInfo details)
                     return details.IsDirty ? nameof(Images.SubmoduleDirty) : nameof(Images.FileStatusModified);
                 }
             }
+
+            private void SetStatusTooltip()
+            {
+                if (Info.Detailed?.RawStatus is not null)
+                {
+                    // Prefer submodule status, shows ahead/behind
+                    TreeViewNode.ToolTipText = LocalizationHelpers.ProcessSubmoduleStatus(
+                        new GitModule(Info.Path),
+                        Info.Detailed.RawStatus,
+                        moduleIsParent: false,
+                        limitOutput: true);
+                }
+                else if (GitStatus is not null)
+                {
+                    var changeCount = new ArtificialCommitChangeCount();
+                    changeCount.Update(GitStatus);
+                    TreeViewNode.ToolTipText = changeCount.GetSummary();
+                }
+                else
+                {
+                    Debug.Assert(true, "Unexpected default status for SetStatusTooltip");
```

Comment:
Nit: I think `Debug.Assert(true, ...)` does nothing at all.
Will be hidden by if all commits are squashed because replaced in a later commit.

Reply:
This was mostly intended as a comment but should have been false...
Intentionally removed.  The first change to RepoObjectsTree.Nodes.Submodules.cs could be separated, then this Assert has some meaning. The changes should be squashed though.